src/NuGet.Clients/NuGet.Console/OutputConsole/OutputConsoleProvider.cs

```diff
@@ -65,12 +66,25 @@ public class OutputConsoleProvider : IOutputConsoleProvider
                         return new OutputConsole(vsOutputWindow, vsUIShell);
                     }
                 }, NuGetUIThreadHelper.JoinableTaskFactory);
+
+            _isServerMode = new AsyncLazy<bool>(
```

Comment:
`IOutputConsoleProvider` is imported in at least one component who's MEF registration is `Shared` (meaning it's a singleton), and it's also used in PMC, whose lifetime is unrelated to solutions.

Therefore, I believe this class needs to listen to solution open/close events, and reset both `_isServerMode` and `cachedOutputConsole`, so that when a customer opens/closes solutions and transitions between online and local scenarios, the output works correctly.

Reply:
There's nothing solution specific in this class. 
A context like is server mode is determined at process startup.

Reply:
So, if you're online and you want to change solutions to a local solution, it's not possible to switch in the same process? You must close the current process and start a new process?

Reply:
Server only. 
client side you can switch between online environments and VS easily.